src/mlpack/core/optimizers/sgd/sgd_impl.hpp

```diff
@@ -109,8 +116,17 @@ double SGD<UpdatePolicyType>::Optimize(
     else
       function.Gradient(iterate, currentFunction, gradient);
 
-    // Time to do something really dirty.
-    gradient.transform( [](double val) { return std::min(std::max(val, -3.), 3.); } );
+    // Clip the gradient.
+    if (clipGradient)
+    {
+      gradient.transform
+      (
+        [&](double val)
+        {
+          return std::min(std::max(val, minGradient), maxGradient);
```

Comment:
Should we implement the gradient clipping as a policy, as done e.g. for momentum? That way other optimization methods could reuse the code? Also, does everybody agree, with implementing gradient clipping inside the optimizer class (e.g. SGD class), I think we could also do this inside the RNN or FFN class.